src/audio/google/google_ctc_audio_processing.c

```diff
@@ -367,10 +352,16 @@ static int ctc_prepare(struct processing_module *mod,
 	}
 	cd->next_avail_output_samples = cd->chunk_frames * num_channels;
 
+	config = comp_get_data_blob(cd->tuning_handler, &config_size, NULL);
+
 	cd->state = GoogleCtcAudioProcessingCreateWithConfig(cd->chunk_frames,
 							     audio_stream_get_rate(&source->stream),
-							     /*config=*/NULL,
-							     /*config_size=*/0);
+							     config,
```

Comment:
Here should be better having the config size check (as you did in ctc_set_config()) before calling GoogleCtcAudioProcessingCreateWithConfig() in case a bad config blob is inhabiting. Consider to replace with {NULL, 0} for creating CTC under that circumstance.

Reply:
The config size is checked in `GoogleCtcAudioProcessingCreateWithConfig `. The config will be ignored if config size is not expected. I agree that checking again here would make it more clear.